### PR TITLE
Added 8-bit Quantization support for RELU_N1_TO_1.

### PR DESCRIPTION
**Pull request by amitsrivastava78**
Added the quantization support for the operator.

### CODE REVIEW

File: tensorflow/lite/kernels/activations.cc [hunk 1]
```diff
@@ -347,6 +347,26 @@ TfLiteStatus ReluEval(TfLiteContext* context, TfLiteNode* node) {
...
 }
 
+namespace {
+template <typename T>
+void QuantizedRelu1(const TfLiteTensor* input, TfLiteTensor* output) {
+  ActivationParams params;
+  int32 kMin = -1;
+  int32 kMax = 1;
+  params.activation_type = FusedActivationFunctionType::kRelu1;
+  // Relu1 has a min raneg of -1, we need to quantize this
```
Comment (suharshs) [inline tensorflow/lite/kernels/activations.cc]: "raneg" -> "range"

File: tensorflow/lite/kernels/activations.cc [hunk 2]
```diff
@@ -347,6 +347,26 @@ TfLiteStatus ReluEval(TfLiteContext* context, TfLiteNode* node) {
...
+  params.quantized_activation_min =
+      (kMin) / input->params.scale + input->params.zero_point;
+  // Relu1 has a max raneg of 1, we need to quantize this
+  params.quantized_activation_max =
+      (kMax) / input->params.scale + input->params.zero_point;
+
+  // Reused the optimzed function written for ReluX
```
Comment (suharshs) [inline tensorflow/lite/kernels/activations.cc]: "optimzed" -> "optimized"

File: tensorflow/lite/kernels/activations.cc [hunk 3]
```diff
@@ -361,8 +381,18 @@ TfLiteStatus Relu1Eval(TfLiteContext* context, TfLiteNode* node) {
...
+      return kTfLiteOk;
+    } break;
+    case kTfLiteInt8: {
+      QuantizedRelu1<int8_t>(input, output);
+      return kTfLiteOk;
+    } break;
     default:
-      context->ReportError(context, "Only float32 supported currently, got %s.",
+      context->ReportError(context,
+                           "Only float32, kTfLiteUInt8, kTfLiteInt8 supported "
```
Comment (suharshs) [inline tensorflow/lite/kernels/activations.cc]: please use uint8 and int8 instead of kTfLiteUint8.

File: tensorflow/lite/kernels/activations.cc [hunk 4]
```diff
@@ -347,6 +347,26 @@ TfLiteStatus ReluEval(TfLiteContext* context, TfLiteNode* node) {
...
+namespace {
+template <typename T>
+void QuantizedRelu1(const TfLiteTensor* input, TfLiteTensor* output) {
+  ActivationParams params;
+  int32 kMin = -1;
+  int32 kMax = 1;
+  params.activation_type = FusedActivationFunctionType::kRelu1;
+  // Relu1 has a min raneg of -1, we need to quantize this
+  params.quantized_activation_min =
+      (kMin) / input->params.scale + input->params.zero_point;
```
Comment (suharshs) [inline tensorflow/lite/kernels/activations.cc]: These quantized_activation_min and quantized_activation_max values are used to clamp the output integer value. To be safe we should also ensure that these values are within the int32 range (the type of quantized_activation_min).
So maybe add something like what is done for QuantizedRelu6 in this file (clamping to the numerical limits )

**Review by suharshs — CHANGES_REQUESTED** [review]

File: tensorflow/lite/kernels/activations.cc (hunk 1 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/activations.cc]: @suharshs , thanks for the review, i have updated the code as per your comments, kindly check and approve.

Regards
Amit

File: tensorflow/lite/kernels/activations.cc (hunk 3 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/activations.cc]: @suharshs , thanks for the review, i have updated the code as per your comments, kindly check and approve.

Regards
Amit

File: tensorflow/lite/kernels/activations.cc (hunk 4 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/activations.cc]: @suharshs , thanks for the review, i have updated the code as per your comments, kindly check and approve.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@suharshs , thanks for the review, i have updated the code as per your comments, kindly check and approve.

Regards
Amit

**Review by suharshs — DISMISSED** [review]

**Comment by amitsrivastava78** [comment]
@rthadur , resolved the merge conflicts
@suharshs , sorry for the trouble i resolved the merge conflicts can you please approve again.


Regards
Amit

**Review by suharshs — DISMISSED** [review]

**Comment by amitsrivastava78** [comment]
@rthadur , can you please help to get this PR merged, this is already approved.

Regards
Amit

**Comment by rthadur** [comment]
> @rthadur , can you please help to get this PR merged, this is already approved.
> 
> Regards
> Amit

sure , @suharshs can you please approve this internally

**Comment by amitsrivastava78** [comment]
> > @rthadur , can you please help to get this PR merged, this is already approved.
> > Regards
> > Amit
> 
> sure , @suharshs can you please approve this internally

@suharshs , can you pls approve this internally as well.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@suharshs , there was some merge conflict which i resolved, can you please approve it again. 

Sorry for the trouble caused.

Regards
Amit

**Review by suharshs — APPROVED** [review]
reapproved, thanks!

**Pull request merged**